### DISCUSSION TITLE
Component Testing: Jest Syntax Error Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by Roland-Hufnagel** [neuefische/questions#315] [Web] [answered]
Ich habe diesen Syntax-Error beim Component Testing. Es liegt wohl an der nanoid, die ich in der Komponente verwende. Diese scheint einen Import zu machen, den Jest nicht unterstützt. Mit Math.random().toString() statt mit nanoid() funktioniert es nämlich. Aber das kann ja nicht die Lösung sein. Hat jemand eine Idee, wie man dieses Problem lösen kann?

![Bildschirmfoto 2022-11-10 um 18 29 59](https://user-images.githubusercontent.com/111574974/201166435-cfb5a170-15f7-46da-8417-fd18a050cf4e.png)

**Comment by ThomasF85** [comment, marked as answer]
Hi Roland,

you can mock nanoid for your tests, which basically means that you give a fake version of nanoid to your test framework. This fake version of nanoid will only be used in tests and in development and production the real nanoid package will be used. For this you have to create a folder `__mocks__` in your project root folder and create a new file `nanoid.js` with a fake nanoid function inside, like this:
```
module.exports = {
  nanoid: () => Math.random().toString(36).substring(2),
};
```

**Reply by Roland-Hufnagel** [reply]
This solution works fine. Thanks a lot!
